components/net/http_cache.rs

```diff
@@ -187,19 +180,34 @@ fn response_is_cacheable(metadata: &Metadata) -> bool {
 fn get_response_expiry_from_headers(headers: &Headers) -> Duration {
     // Calculating Freshness Lifetime https://tools.ietf.org/html/rfc7234#section-4.2.1
     if let Some(&header::CacheControl(ref directives)) = headers.get::<header::CacheControl>() {
-        for directive in directives {
-            match directive {
-                &header::CacheDirective::SMaxAge(secs) => {
-                    return Duration::seconds(secs as i64);
-                },
-                &header::CacheDirective::MaxAge(secs) => {
-                    return Duration::seconds(secs as i64);
-                },
-                _ => (),
+        let has_no_cache_directive = directives.iter().any(|directive| {
+            if let header::CacheDirective::NoCache = *directive {
```

Comment:
Why does `header::CacheDirective::NoCache == *directive` not work?

Reply:
it does work, thanks!